### PR TITLE
www/caddy: Changes for version 1.5.3 and 1.5.4

### PR DESCRIPTION
**Pull request by Monviech**
www/caddy: Changes for version 1.5.3

Reference: https://github.com/opnsense/plugins/pull/3865

www/caddy: Changes for version 1.5.4

Reference: https://github.com/opnsense/plugins/pull/3891

### CODE REVIEW

**Comment by Monviech** [comment]
@AdSchellevis This one is finished and can be merged, thank you a lot. ^^

File: source/manual/how-tos/caddy.rst [hunk 1]
```diff
@@ -347,28 +386,121 @@ Reverse proxy the OPNsense WebUI
...
+
+.. Note:: Open ``https://opn.example.com`` and it should serve the reverse proxied OPNsense WebUI. Check the log file for errors if it doesn't work, most of the time the TLS Server Name doesn't match the SAN of the `TLS Trusted CA Certificate`. Caddy doesn't support CN (Common Name) in certificate since it's been deprecated since many years. Only SAN certificates work.
+.. Attention:: Create an access list to restrict access to the WebUI. Add that access list to this domain.
+
+
+-------------------------------
+Redirect ACME HTTP-01 challenge
+-------------------------------
+
+Sometimes an application behind Caddy uses it's own ACME Client to get certificates, most likely with the HTTP-01 challenge. This plugin has a built in mechanism to redirect this challenge type easily to a destination behind it.
```
Comment (dmarcoux) [inline source/manual/how-tos/caddy.rst:417]: Small typo:

```suggestion
Sometimes an application behind Caddy uses its own ACME Client to get certificates, most likely with the HTTP-01 challenge. This plugin has a built in mechanism to redirect this challenge type easily to a destination behind it.
```

File: source/manual/how-tos/caddy.rst [hunk 2]
```diff
@@ -347,28 +386,121 @@ Reverse proxy the OPNsense WebUI
...
+**Upstream Domain:**                ``192.168.10.1``
+**Upstream Port:**                  ``443``
+**TLS:**                            ``X``
+**TLS Server Name**:                ``foo.example.com``
+=================================== ============================
+
 * Press **Save** and **Apply**
 
-Go to `System - Settings - Administration`
+.. Note:: With this configuration, Caddy will eventually choose the TLS-ALPN-01 challenge for its own ``foo.example.com`` domain, and reverse proxy the HTTP-01 challenge to ``192.168.10.1``, where the upstream destination can listen on port 80 for ``foo.example.com`` and solve it's own challenge for a certificate. With TLS enabled in the Handler, an encrypted connection is automatically possible. The automatic HTTP to HTTPS redirection is also taken care of.
```
Comment (dmarcoux) [inline source/manual/how-tos/caddy.rst:454]:
```suggestion
.. Note:: With this configuration, Caddy will eventually choose the TLS-ALPN-01 challenge for its own ``foo.example.com`` domain, and reverse proxy the HTTP-01 challenge to ``192.168.10.1``, where the upstream destination can listen on port 80 for ``foo.example.com`` and solve its own challenge for a certificate. With TLS enabled in the Handler, an encrypted connection is automatically possible. The automatic HTTP to HTTPS redirection is also taken care of.
```

**Comment by Monviech** [comment]
@dmarcoux Thank you for looking over it. ^^

**Pull request merged**